contracts/LightClient.sol

```diff
@@ -121,4 +149,63 @@ contract LightClient is ILightClient, Ownable {
     function proposedValidators() external view override returns (address[] memory, uint256[] memory) {
         return staking.proposedValidators();
     }
+
+    function getEpochIdx(uint256 height) public view override returns (uint256) {
+        require(isInHeightRange(height), "out of height range");
+        // Reduce the times of Sload
+        uint256 _epochPeriod = epochPeriod;
+        return (height + _epochPeriod - 1) / _epochPeriod;
+    }
+
+    function isInHeightRange(uint256 height) public view override returns (bool) {
+        (uint256 min, uint256 max) = heightRange();
+        if (height >= min && height <= max) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function minEpochIdx() public view override returns (uint256) {
+        if (curEpochIdx >= TOTAL_EPOCH) {
+            return curEpochIdx - TOTAL_EPOCH + 1;
+        }
+        return 0;
+    }
+
+    function heightRange() public view override returns (uint256 min, uint256 max) {
+        min = _minHeight();
+        max = _maxHeight();
+        // when the latestEpochHeight is 10000 and TOTAL_EPOCH = 2,
+        // the range of the block ,can be verified by contract, is [1,20000]
+    }
+
+    function _minHeight() internal view returns (uint256) {
+        uint256 _minEpochId = minEpochIdx();
+        if (_minEpochId == 0) {
+            return 0;
+        } else {
+            return (_minEpochId - 1) * epochPeriod + 1;
+        }
+    }
+
+    function _maxHeight() internal view returns (uint256) {
+        return curEpochHeight() + epochPeriod;
+    }
+
+    function _deriveEpochHeight(uint256 _epochIdx, uint256 _epochPeriod) internal pure returns (uint256) {
+        require(_epochIdx != 0, "epochIdx can not be 0");
+        uint256 _curEpochHeight = (_epochIdx - 1) * _epochPeriod;
+        return _curEpochHeight;
+    }
+
+    function _deriveEpochIdx(uint256 _epochHeight, uint256 _epochPeriod) internal pure returns (uint256) {
+        require(_epochHeight % _epochPeriod == 0, "invalid epochHeight");
+        uint256 _epochIdx = _epochHeight / _epochPeriod + 1;
```

Comment:
I think the function here should be well documented - is it supposed to find the epoch header (i.e., the last block of the previous epoch) or others?